### PR TITLE
MultiMatch Full Demo

### PR DESCRIPTION
**Pull request by kev-chien**
A semester's worth of work, with 4 iterations worth of features. This is the first PR of code from the Aakup fork to the golden repo.

Key features:
- [x] Users can **create and rename projects**
- [x] Users can **add participants**
- [x] Users can **add project times**
- [x] Users can **send emails to participants to rank preferences**
- [x] Participants **can edit their preferences**
- [x] Users can submit collected rankings to Mark's API to **generate a matching**
- [x] Users can **modify the matching**
- [x] Users can **email out final matchings to participants**

### CODE REVIEW

File: app/controllers/matchings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  #         format.html { redirect_to project_matching_path }
+  #       # else
+  #       #   format.html { render :edit }
+  #       end
+  #     end
+  #   end
+  # end
+
+  # POST /projects/:project_id/matching
+  def create
```
Comment (adnanhemani) [inline app/controllers/matchings_controller.rb:72]: Please put most of the logic into the model instead of cthe controller

**Comment by adnanhemani** [comment]
Running the rake:db commands errors out for some reason on seeding the db, but rake db:reset works. Please investigate and update README and/or db code as needed.

**Comment by adnanhemani** [comment]
In your app, when clicking the dropdown menu next to the username, it has two buttons that don't do anything: "Something" and "Settings". Either rename the button or take them out.

**Review by adnanhemani — CHANGES_REQUESTED** [review]

**Comment by kev-chien** [comment]
> Running the rake:db commands errors out for some reason on seeding the db, but rake db:reset works. Please investigate and update README and/or db code as needed.

Addressed in [this commit](https://github.com/saasbook/MultiMeet/pull/17/commits/624d9a3da4c9792f56b6bd67f6abe1232ff67263)

**Comment by kev-chien** [comment]
> In your app, when clicking the dropdown menu next to the username, it has two buttons that don't do anything: "Something" and "Settings". Either rename the button or take them out.

Addressed in [this commit](https://github.com/saasbook/MultiMeet/pull/17/commits/c862b7e86df5cb9ce4c52df04ac4128991735051)

File: app/controllers/matchings_controller.rb (hunk 1 above)
Comment (kev-chien) [reply app/controllers/matchings_controller.rb:72]: Addressed in [this commit](https://github.com/saasbook/MultiMeet/pull/17/commits/34f23ba6d1051169f3d52cf51e36f8eec3688648)

**Comment by kev-chien** [comment]
@adnanhemani we think we've addressed the changes requested, could you look into it for us please? Thanks!

**Comment by adnanhemani** [comment]
LGTM

**Pull request merged**